### PR TITLE
Replacing dhttpd with a shelf server

### PR DESCRIPTION
**Pull request by domesticmouse**
This custom server adds CORS headers to all responses for local development. With https://github.com/dart-lang/dart-services/pull/391 we have a working system, at least for local development.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -25,10 +25,11 @@ dev_dependencies:
...
   build_web_compilers: any
   collection: ^1.14.10
-  dhttpd: any
   git: any
   grinder: ^0.8.0
   pedantic: ^1.3.0
   test: ^1.3.4
   tuneup: any
   yaml: any
+  shelf: ^0.7.5
```
Comment (devoncarew) [inline pubspec.yaml]: nit: I'd sort these lexically

File: web/scripts/assets/FontManifest.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (devoncarew) [inline web/scripts/assets/FontManifest.json]: I understand why this is here, but on the face of it it's a bit mysterious. Can we doc why this is here somewhere?

Comment (kevmoo) [reply web/scripts/assets/FontManifest.json]: required. can discuss offline

Comment (devoncarew) [reply web/scripts/assets/FontManifest.json]: Yes, I understand the reason, but it would be useful to doc somewhere why this empty file is here.

**Pull request merged**